lib/async/io/endpoint.rb

```diff
@@ -59,10 +60,23 @@ def accept(backlog = Socket::SOMAXCONN, &block)
 				end
 			end
 			
+			# Create an Endpoint instance by URI scheme
+			#
+			# The host and port of the URI will be passed to the Endpoint
+			# factory method, along with any options.
+			#
+			# @param string [String] URI as string. Scheme will decide implementation used.
+			# @param options keyword arguments passed through to {#initialize}
+			#
+			# @see Endpoint.socket socket - invoked when parsing a URL with the socket scheme "socket://127.0.0.1"
```

Comment:
Should we roll with this to start with, or should we try to keep it idealistic?

e.g.

@see Endpoint.socket Invoked when parsing a URL with the socket scheme "socket://127.0.0.1"

Love what you've done here.